### PR TITLE
Use anonymous consts in expansion

### PR DESCRIPTION
**Pull request by Veykril**
Since this now depends on syn 2.0, the MSRV was raised beyond the addition of this feature anyways, and with this change rust-analyzer can finally drop the special casing for this crate's const item impl hack.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -2488,14 +2469,8 @@ impl<'a> Structure<'a> {
                     const _: () = { #generated };
                 })
             } else {
-                let dummy_const: Ident = sanitize_ident(&format!(
```
Comment (mystor) [inline src/lib.rs:2491]: WE already have a config (`self.underscore_const`) which implements this feature. We should remove that config and make the existing setter method a no-op, simplifying the code.

**Review by mystor — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (Veykril) [reply src/lib.rs:2491]: Huh, I was tunnel visioning very hard on this one sorry 😅

**Comment by Veykril** [comment]
Noticed this PR was still open, so a friendly ping @mystor

**Comment by Urgau** [comment]
@mystor `rustc` is now in the process of adding a lint against non-local impl definitions, see https://github.com/rust-lang/rust/pull/120393.
It would be great to get this merged since `rustc` itself is having issue with this crate.

**Review by mystor — APPROVED** [review]

**Pull request merged**